runtime/src/lib.rs

```diff
@@ -1331,6 +1410,50 @@ impl_runtime_apis! {
         }
     }
 
+    impl pallet_ismp_runtime_api::IsmpRuntimeApi<Block, <Block as BlockT>::Hash> for Runtime {
+        fn host_state_machine() -> StateMachine {
+            <Runtime as pallet_ismp::Config>::HostStateMachine::get()
+        }
+
+        fn challenge_period(id: StateMachineId) -> Option<u64> {
+            pallet_ismp::Pallet::<Runtime>::challenge_period(id)
+        }
+
+        fn generate_proof(
+            keys: ProofKeys
+        ) -> Result<(Vec<Leaf>, Proof<<Block as BlockT>::Hash>), sp_mmr_primitives::Error> {
+            pallet_ismp::Pallet::<Runtime>::generate_proof(keys)
```

Comment:
I don't get how this possibly works if we set Mmr = NoOpMmmr in the ismp pallet config

Reply:
This is the description of the `Mmr` type from pallet_ismp

```
// Optional merkle mountain range overlay tree, for cheaper outgoing request proofs.
// You most likely don't need it, just use the `NoOpMmrTree`
```

That being said, I'll double check with them  if we need it